src/redfishpower/plugs.c

```diff
@@ -103,6 +103,15 @@ void plugs_remove(plugs_t *p, const char *plugname)
     hostlist_delete(p->plugs, plugname);
 }
 
+void plugs_set_plug_sub_required_flag(plugs_t *p,
+                                      const char *plugname,
+                                      int plug_sub_required)
+{
+    struct plug_data *pd = plugs_get_data(p, plugname);
+    if (pd)
+        pd->plug_sub_required = plug_sub_required;
+}
+
```

Comment:
What's the reason for this flag?  It's not clear from the description.

Seems like `{{plug}}` should be subtituted if there, and not if not, and that's that?

Reply:
I'll add some comments to clarify this more.  But basically before this PR number of plugs == number of indices of hosts.  i.e.

`setplugs Plugs[0-7] [0-7]`

but now that we can do plug substitution, multiple can now point to just 1

`setplugs Plugs[0-7] 0`

but if you have > 1 pointing to 1, then plug substitution is required.  So it's just a flag to say "hey, this plug better be setup with a path with plug substitution".  

The reason the flag is necessary is because we setup paths separately, so there's no way to check for this condition during a `setplugs` call.

Reply:
> Gotcha, although it might be more straightforward to simply check the whole config at the end, ensuring that every plug maps to a unique (host,path) tuple for each command, e.g. after any substitution.

I've gone back and forth about that.  Like maybe the first time someone does a stat/on/off, do a "is config legit?" kinda check?  But I feel that nothing should stop me from performing a stat/on/off just because some other host is messed up config wise.

Perhaps we could add a "do_final_check" command that we add at the end of the `login` device script?